src/__init__.py

```diff
@@ -121,26 +129,55 @@ def fputs(string):
             putchar(' ')
             col += 1
         while col <= colright:
-            if numlines == 6:
-                char = background6[lin][col]
-            elif numlines == 18:
-                char = background18[lin][col]
-            elif numlines == 19:
-                char = background19[lin][col]
-            elif numlines == 21:
-                char = background21[lin][col]
-            elif numlines == 22:
-                char = background22[lin][col]
-            elif numlines == 23:
-                char = background23[lin][col]
-            elif numlines == 24:
-                char = background24[lin][col]
-            elif numlines == 29:
-                char = background29[lin][col]
-            elif numlines == 32:
-                char = background32[lin][col]
+            if hemisphere == 'north':
+                # north - read moons from upper-left to bottom-right
+                if numlines == 6:
+                    char = background6[lin][col]
+                elif numlines == 18:
+                    char = background18[lin][col]
+                elif numlines == 19:
+                    char = background19[lin][col]
+                elif numlines == 21:
+                    char = background21[lin][col]
+                elif numlines == 22:
+                    char = background22[lin][col]
+                elif numlines == 23:
+                    char = background23[lin][col]
+                elif numlines == 24:
+                    char = background24[lin][col]
+                elif numlines == 29:
+                    char = background29[lin][col]
+                elif numlines == 32:
+                    char = background32[lin][col]
+                else:
+                    char = '@'
             else:
-                char = '@'
+                # south - read moons from bottom-right to upper-left
+                # equivalent to rotate 180 degress or turn upside-down
+                if numlines == 6:
+                    char = background6[-1-lin][-col-1]
```

Comment:
this line fixes the 6 line moon beeing incorrectly displayed, but I want to further improve or review why it was needed to shift the column.